cmd/release-controller/sync_verify.go

```diff
@@ -28,19 +28,30 @@ func (c *Controller) ensureVerificationJobs(release *Release, releaseTag *imagev
 				}
 			}
 
+			jobAttempt := 0
 			if status, ok := verifyStatus[name]; ok {
+				jobAttempt = status.Retries
 				switch status.State {
-				case releaseVerificationStateFailed, releaseVerificationStateSucceeded:
-					// we've already processed this, continue
+				case releaseVerificationStateSucceeded:
 					continue
+				case releaseVerificationStateFailed:
+					jobAttempt++
+					if verifyType.Optional || jobAttempt >= releaseVerificationBlockingJobAttempts {
+						continue
+					}
 				case releaseVerificationStatePending:
 					// we need to process this
 				default:
 					glog.V(2).Infof("Unrecognized verification status %q for type %s on release %s", status.State, name, releaseTag.Name)
 				}
 			}
 
-			job, err := c.ensureProwJobForReleaseTag(release, name, verifyType, releaseTag)
+			jobName := name
+			if jobAttempt > 0 {
+				jobName = fmt.Sprintf("%s-%d", name, jobAttempt)
```

Comment:
Hrm, I think we may want to consider instead deleting the existing prow job if it is for an older attempt.  There are advantages to leaving the old jobs around, but also downsides.  Let me think a bit about it.  Either way, I would rather push job name calculation into ensureProwJobForReleaseTag (pass attempt)

Reply:
I guess I'm ok with this structure.  Just move it inside ensureProwJobForReleaseTag and pass `jobAttempt` down